core/src/main/java/org/infinispan/commands/statetransfer/AbstractStateTransferCommand.java

```diff
@@ -0,0 +1,78 @@
+package org.infinispan.commands.statetransfer;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+
+import org.infinispan.commands.TopologyAffectedCommand;
+import org.infinispan.commands.remote.BaseRpcCommand;
+import org.infinispan.commons.util.IntSet;
+import org.infinispan.remoting.transport.Address;
+import org.infinispan.util.ByteString;
+
+/**
+ * Base class for commands related to state transfer.
+ *
+ * @author Ryan Emerson
+ * @since 11.0
+ */
+abstract class AbstractStateTransferCommand extends BaseRpcCommand implements TopologyAffectedCommand {
+
+   private byte commandId;
```

Comment:
[opinion]
I'm not sure about this anymore :) we are allocating +4 bytes per instance just to avoid implementing `byte getCommandId()`

Reply:
Why +4? Is it not just  a single byte per instance?

I'm against removing it though

Reply:
1 byte + 3 bytes alignment. 
```
org.infinispan.commands.statetransfer.AbstractStateTransferCommand object internals:
 OFFSET  SIZE         TYPE DESCRIPTION                               VALUE
      0    12              (object header)                           N/A
     12     4   ByteString BaseRpcCommand.cacheName                  N/A
     16     4      Address BaseRpcCommand.origin                     N/A
     20     4          int AbstractStateTransferCommand.topologyId   N/A
     24     1         byte AbstractStateTransferCommand.commandId    N/A
     25     3              (alignment/padding gap)                  
     28     4       IntSet AbstractStateTransferCommand.segments     N/A
Instance size: 32 bytes
Space losses: 3 bytes internal + 0 bytes external = 3 bytes total
```

Reply:
> I'm against removing it though

Honestly there was meant to be a *not* in there :laughing: 

Ok, I'll remove :+1:

Reply:
actually, I may be wrong. I think the JVM has to align to 32 bytes anyway...